### ISSUE TITLE
Population out of sync causes Panic and crash

### ISSUE THREAD

**Issue by dbryar** [pingidentity/terraform-provider-pingone#398]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### PingOne Terraform provider Version
0.14

### Terraform Version
1.4.2

### Affected Resource(s)
- pingone_population

### Panic Output
```log
Stack trace from the terraform-provider-pingone_v0.14.0 plugin:
panic: interface conversion: interface {} is nil, not *management.Population
goroutine 486 [running]:
github.com/pingidentity/terraform-provider-pingone/internal/service/sso.PingOnePopulationRead({0x1649ca0?, 0xc000acd9b0}, 0xc000266000, {0xc0006424b0, 0x24}, {0xc0006424e0, 0x24})
	github.com/pingidentity/terraform-provider-pingone/internal/service/sso/resource_population.go:229 +0x193
github.com/pingidentity/terraform-provider-pingone/internal/service/sso.resourcePingOnePopulationRead({0x1649c68, 0xc000acd920}, 0xc00053ac00, {0x121a2c0?, 0xc000478250})
	github.com/pingidentity/terraform-provider-pingone/internal/service/sso/resource_population.go:98 +0x247
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0003968c0, {0x1649ca0, 0xc000acd2f0}, 0xd?, {0x121a2c0, 0xc000478250})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:724 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0003968c0, {0x1649ca0, 0xc000acd2f0}, 0xc000553450, {0x121a2c0, 0xc000478250})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc00012f410, {0x1649ca0?, 0xc000acd170?}, 0xc000056980)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:[613](https://gitlab.com/b707/platform-and-payments/pinocchio/-/jobs/4421820986#L613) +0x4a5
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ReadResource({{0x1651a50?, 0xc00012f410?}}, {0x1649ca0?, 0xc000acd170?}, 0xc000056800?)
	github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf5to6server/tf5to6server.go:105 +0x242
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.muxServer.ReadResource({0xc000371e30, 0xc000371e90, {0xc00070eca0, 0x2, 0x2}, {0x0, 0x0, 0x0}, {0x0, 0x0, ...}, ...}, ...)
	github.com/hashicorp/terraform-plugin-mux@v0.10.0/tf6muxserver/mux_server_ReadResource.go:26 +0x102
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc000528e60, {0x1649ca0?, 0xc000acc420?}, 0xc00029a7e0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov6/tf6server/server.go:746 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x1415e80?, 0xc000528e60}, {0x1649ca0, 0xc000acc420}, 0xc0002f43f0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.15.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:383 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00021a000, {0x1650238, 0xc000602340}, 0xc0006d9440, 0xc0003dd8c0, 0x1f28d10, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1345 +0xdf3
google.golang.org/grpc.(*Server).handleStream(0xc00021a000, {0x1650238, 0xc000602340}, 0xc0006d9440, 0x0)
	google.golang.org/grpc@v1.54.0/server.go:1722 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.54.0/server.go:966 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.54.0/server.go:964 +0x28a
Error: The terraform-provider-pingone_v0.14.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Steps to Reproduce
A population was manually (though UI) deleted and the TF state became out of sync

### Important Factoids
- I have tried to recreate the same populations (manually) and import with same resource names
- I have tried deleting all populations in state (at least one must remain in Ping One)

**Comment by dbryar** [comment]
@patrickcping I've backtraced the logs and it looks like the root cause was a name change to a population in the HCL. The provider tried to delete the existing population (instead of changing the name) and recreate it, but the population had users in it so the operation failed.

The users were removed manually, and the operation repeated, and then on the next config change the panic during the `plan` occurred

**Comment by dbryar** [comment]
Further, the resource name in HCL is based on the object name, so I can see why it tried to delete and recreate. As far as HCL was concerned they were two different resources

**Comment by patrickcping** [comment]
@dbryar ok this makes sense - with the same resource name the population should update it's name without needing a replace plan.  With different resource names we're left to the Terraform plan hierarchy to work out what to do.  This might help though: https://developer.hashicorp.com/terraform/cli/commands/state/mv#example-rename-a-resource

I've released `v0.15.1` today which solves the panic lock

**Comment by dbryar** [comment]
confirmed fix 🥳
